src/main/scala/inox/ast/SymbolOps.scala

```diff
@@ -115,6 +115,44 @@ trait SymbolOps { self: TypeOps =>
     case _ => false
   }
 
+  /** Returns true when the given expression is ground, pure, and does not contain
+   *  choose or quantifiers, and can thus be simplified.
+   */
+  def canBeForced(expr: Expr): Boolean = {
+    isGround(expr) && isPure(expr)(PurityOptions.AssumeChecked) && !containsChooseOrQuantifiers(expr)
```

Comment:
Note that you have to be careful about transitive function calls that contain chooses or quantifiers. A more precise approach would be to add an option to the evaluators to crash when encountering such an expression. Otherwise, you can compute the fixpoint of all transitive callees that contain chooses or quantifiers, but you will probably miss many opportunities to simplify in doing so since you'll have to consider code that wouldn't be evaluated given the concrete ground arguments.

Also, you can't use `AssumeChecked` here if it hasn't been enabled in the context. It would be better to take an implicit `PurityOptions` argument like `simplifyExpr`.

On a general design note, I would not declare this function outside of `simplifyGround`. The symbols get imported in many places so we're trying to avoid polluting the namespace unless it's either a useful override point or a useful standalone function.

Reply:
> Note that you have to be careful about transitive function calls that contain chooses or quantifiers. A more precise approach would be to add an option to the evaluators to crash when encountering such an expression. Otherwise, you can compute the fixpoint of all transitive callees that contain chooses or quantifiers, but you will probably miss many opportunities to simplify in doing so since you'll have to consider code that wouldn't be evaluated given the concrete ground arguments.

Ok, I will go with the evaluator option then. Thanks.

> Also, you can't use AssumeChecked here if it hasn't been enabled in the context. It would be better to take an implicit PurityOptions argument like simplifyExpr.

Right, fixed! (commits incoming)

> On a general design note, I would not declare this function outside of simplifyGround. The symbols get imported in many places so we're trying to avoid polluting the namespace unless it's either a useful override point or a useful standalone function.

Of course, fixed as well.